portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -45,23 +56,127 @@ public void tearDown() {
+    new DataServlet().doPost(postRequest, postResponse);
+    
+    Assert.assertEquals(1, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+  }
+
+  @Test 
+  public void testPOSTMultipleComments() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    HttpServletRequest postRequest = mock(HttpServletRequest.class);
+    HttpServletResponse postResponse = mock(HttpServletResponse.class);
+
+    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+
+    when(postRequest.getParameter("text-input")).thenReturn("Comment1");
+    new DataServlet().doPost(postRequest, postResponse);
+    when(postRequest.getParameter("text-input")).thenReturn("Comment2");
+    new DataServlet().doPost(postRequest, postResponse);
+    when(postRequest.getParameter("text-input")).thenReturn("Comment3");
+    new DataServlet().doPost(postRequest, postResponse);
+    
+    Assert.assertEquals(3, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+  }
+
+  @Test 
+  public void testPOSTNull() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    HttpServletRequest postRequest = mock(HttpServletRequest.class);
+    HttpServletResponse postResponse = mock(HttpServletResponse.class);
+
+    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+
+    when(postRequest.getParameter("text-input")).thenReturn(null);
+    new DataServlet().doPost(postRequest, postResponse);
+    
+    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void testGETSingleComment() throws Exception {
+    HttpServletRequest postRequest = mock(HttpServletRequest.class);
+    HttpServletResponse postResponse = mock(HttpServletResponse.class);
+    HttpServletRequest getRequest = mock(HttpServletRequest.class);
+    HttpServletResponse getResponse = mock(HttpServletResponse.class);
+
+    when(postRequest.getParameter("text-input")).thenReturn("Test Comment #1");
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(getResponse.getWriter()).thenReturn(printWriter);
+
+    new DataServlet().doPost(postRequest, postResponse);
+    new DataServlet().doGet(getRequest, getResponse);
+
+    printWriter.flush(); //may not have flushed yets
+    System.out.println(stringWriter);
+    assertThat(stringWriter.toString()).contains("Test Comment #1");
+  }
+
   @Test
-  public void testJSONConverter() throws Exception {
-    HttpServletRequest requestPost = mock(HttpServletRequest.class);
-    HttpServletResponse responsePost = mock(HttpServletResponse.class);
-    HttpServletRequest requestGet = mock(HttpServletRequest.class);
-    HttpServletResponse responseGet = mock(HttpServletResponse.class);
+  public void testGETAllComments() throws Exception {
```

Comment:
nit - do you also want to add a test that verifies that `max-num` works as intended?

Reply:
yes I am doing that right now!